src/DefinitionResolver.php

```diff
@@ -667,12 +666,26 @@ public function resolveExpressionNodeToType($expr)
                 } else {
                     $classFqn = substr((string)$t->getFqsen(), 1);
                 }
-                $fqn = $classFqn . '->' . $expr->memberName->getText($expr->getFileContents());
+                $add = '->' . $expr->memberName->getText($expr->getFileContents());
                 if ($expr->parent instanceof Node\Expression\CallExpression) {
-                    $fqn .= '()';
+                    $add .= '()';
                 }
-                $def = $this->index->getDefinition($fqn);
-                if ($def !== null) {
+                $lookupDefinition = function (string $base, string $add) use (&$lookupDefinition) {
```

Comment:
this is essentially `CompletionProvider::expandParentFqns()`, I would refactor that